src/misc-utils.ts

```diff
@@ -172,3 +172,38 @@ export async function getLinesFromCommand(
   const { stdout } = await execa(command, args, options);
   return stdout.split('\n').filter((value) => value !== '');
 }
+
+/**
+ * Converts the given GitHub repository URL to its HTTPS version.
+ *
+ * A "GitHub repository URL" looks like one of:
+ *
+ * - https://github.com/OrganizationName/RepositoryName
+ * - git@github.com:OrganizationName/RepositoryName.git
+ *
+ * If the URL does not match either of these patterns, an error is thrown.
+ *
+ * @param url - The URL to convert.
+ * @returns The HTTPS URL of the repository, e.g.
+ * `https://github.com/OrganizationName/RepositoryName`.
+ */
+export function convertToHttpsGitHubRepositoryUrl(url: string): string {
```

Comment:
This logic is not in `auto-changelog`, but is in this repo, so we're actually doing extra. I was curious why there was a distinction, so I did a bit of digging:

- In building `create-release-branch`, I took a lot of code from `action-create-release-pr`
- We used to have a separate repo called `action-monorepo-release-pr`, and that got merged into `action-create-release-pr`
- In particular this logic comes from here: https://github.com/MetaMask/action-monorepo-release-pr/commit/af8d7504aa7b8cd4a65a81625839adf3afc3cbb9#diff-b2d6c9ae0590887f3d213032d057be61bd3e628650fb7ba99391bdc84587e454R25

Reply:
Also, I've simplified the original logic to take a full-on regex approach. It seemed that the code wanted to go there.

Reply:
I will probably end up copying this back to `action-create-release-pr` so that it is consistent between the two.